components/merkledb/src/backends/rocksdb.rs

```diff
@@ -228,7 +259,12 @@ impl Database for RocksDB {
 
 impl Snapshot for RocksDBSnapshot {
     fn get(&self, resolved_addr: &ResolvedAddress, key: &[u8]) -> Option<Vec<u8>> {
-        if let Some(cf) = self.db.cf_handle(&resolved_addr.name) {
+        if let Some(cf) = self
+            .db
+            .read()
+            .expect("Couldn't get read lock to DB")
+            .cf_handle(&resolved_addr.name)
```

Comment:
I wanted to suggest to consider using caching `ColumnFamily`s returned by `cf_handle`, but as far as I see, there's no way to obtain an owned `ColumnFamily` (it doesn't implement `Clone`, for example). Is this an intentional restriction? I think there could be *some* performance boost from getting rid of locks here and in similar scenarios.

Reply:
Yeah. I thought about such approach, but lack of clone. And about restriction. Adding a `Clone` implementation for `ColumnFamily` (which contains inside just a pointer to C object) can lead to `using after drop` case. Maybe adding a lifetime to `ColumnFamily` can fix this issue. I will try to fix it in the `rust-rocksdb` crate in the nearest future.